src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -0,0 +1,1088 @@
+                {
+                    return null;
+                }
+
+                // before adding it to cache, let us do the security check 
+                CheckTypeForwardedTo(assm, type.GetTypeInfo().Assembly, type);
+
+                entry = new TypeNAssembly();
+                entry.Type = type;
+                entry.AssemblyName = assemblyName;
+                _typeCache.SetCachedValue(entry);
+            }
+            return entry.Type;
+        }
+
+        private static void GetSimplyNamedTypeFromAssembly(Assembly assm, string typeName, ref Type type)
+        {
+            // Catching any exceptions that could be thrown from a failure on assembly load
+            // This is necessary, for example, if there are generic parameters that are qualified with a version of the assembly that predates the one available
+            try
+            {
+                type = FormatterServices.GetTypeFromAssembly(assm, typeName);
+            }
+            catch (TypeLoadException) { }
+            catch (FileNotFoundException) { }
+            catch (FileLoadException) { }
+            catch (BadImageFormatException) { }
+        }
+
+        private string _previousAssemblyString;
+        private string _previousName;
+        private Type _previousType;
+
+        internal Type GetType(BinaryAssemblyInfo assemblyInfo, string name)
+        {
+            Type objectType = null;
+
+            if (((_previousName != null) && (_previousName.Length == name.Length) && (_previousName.Equals(name))) &&
+                ((_previousAssemblyString != null) && (_previousAssemblyString.Length == assemblyInfo._assemblyString.Length) && (_previousAssemblyString.Equals(assemblyInfo._assemblyString))))
+            {
+                objectType = _previousType;
+            }
+            else
+            {
+                objectType = Bind(assemblyInfo._assemblyString, name);
+                if (objectType == null)
+                {
+                    Assembly sourceAssembly = assemblyInfo.GetAssembly();
+
+                    if (_isSimpleAssembly)
+                    {
+                        GetSimplyNamedTypeFromAssembly(sourceAssembly, name, ref objectType);
+                    }
+                    else
+                    {
+                        objectType = FormatterServices.GetTypeFromAssembly(sourceAssembly, name);
+                    }
+
+                    // here let us do the security check 
+                    if (objectType != null)
+                    {
+                        CheckTypeForwardedTo(sourceAssembly, objectType.GetTypeInfo().Assembly, objectType);
+                    }
+                }
+
+                _previousAssemblyString = assemblyInfo._assemblyString;
+                _previousName = name;
+                _previousType = objectType;
+            }
+            //Console.WriteLine("name "+name+" assembly "+assemblyInfo.assemblyString+" objectType "+objectType);
```

Comment:
It seems like you caught some of the old commented out debugging code, but not this line.